internal/langserver/handlers/formatting.go

```diff
@@ -14,12 +14,23 @@ import (
 func (h *logHandler) TextDocumentFormatting(ctx context.Context, params lsp.DocumentFormattingParams) ([]lsp.TextEdit, error) {
 	var edits []lsp.TextEdit
 
+	edits, err := formatDocument(ctx, h, params.TextDocument.URI)
+	if err != nil {
+		return edits, err
+	}
+
+	return edits, nil
+}
+
+func formatDocument(ctx context.Context, h *logHandler, uri lsp.DocumentURI) ([]lsp.TextEdit, error) {
```

Comment:
As discussed via Slack yesterday I think it would be better to only extract the minimum amount of logic here, for a couple of reasons:

 - the code action handler will likely need access to the original document and possibly tfExec as well for other future actions we may introduce later
 - smaller functions will less responsibilities are usually easier to read, maintain and test
 - we gain some extra safety with compile-time checks

With that in mind I'd propose something like the following

```go
func formatDocument(tfExec exec.TerraformExecutor, doc filesystem.DocumentHandler) (filesystem.DocumentChanges, error)
```
What do you think?